addons/portal/views/portal_templates.xml

```diff
@@ -833,10 +731,330 @@
+                        <span t-if="not partner.state_id" t-out="partner.country_id.name" class="d-block small"/>
+                    </div>
+                </div>
+                <div class="mt-3">
+                    <a t-attf-href="/portal/address?address_type={{partner.type}}&amp;partner_id={{partner.id}}&amp;portal_address=True"
+                        class="btn btn-link p-0"
+                        role="button"
+                        title="Edit this address"
+                        aria-label="Edit this address">
+                        Edit
+                    </a>
+                    <span id="delete-button" t-att-hidden="partner.type == 'contact'" t-att-style="'display: none;' if selected else ''">
+                        <span class="mx-2">|</span>
+                        <a t-attf-href="/archive/address/#{partner.id}"
+                            class="btn btn-link text-danger p-0"
+                            role="button"
+                            title="Archive this address"
+                            aria-label="Archive this address">
+                            Delete
+                        </a>
+                    </span>
+                    <span id="default-button" t-att-style="'display: none;' if selected else ''">
+                        <span class="mx-2">|</span>
+                        <a class="btn btn-link p-0 js_set_default"
+                           role="button"
+                           title="Default this address"
+                           aria-label="Default this address"
+                           t-att-data-mode="mode"
+                           t-att-data-partner-id="partner.id">
+                            Set as default
+                        </a>
+                    </span>
+                </div>
+            </div>
+        </div>
+    </template>
+
+    <template id="portal.portal_my_details" name="Address Management">
+        <t t-set="no_footer" t-value="1"/>
+        <t t-call="portal.portal_layout">
+            <div id="wrap">
+                <div class="o_portal_address_fill">
+                    <div class="row">
+                        <div class="oe_cart col-12 col-lg-8">
+                            <div>
+                                <t t-if="partner_sudo.type == 'invoice'">
+                                    <h3 class="mb-3">Billing address</h3>
+                                </t>
+                                <t t-elif="partner_sudo.type == 'delivery'">
+                                    <h3 class="mb-3">Delivery address</h3>
+                                </t>
+                                <t t-else="">
+                                    <h3 class="mb-3">Address</h3>
+                                </t>
+                                <div id="errors"/> <!-- for js -->
+                                <form
+                                    action="/portal/address/submit?portal_address=True"
```

Comment:
Is the query parameter here intended? If yes, the JS request should probably use form.action to reflect that